### PR TITLE
cloudapi: format error string in the depsolving

### PR DESCRIPTION
**Pull request by lavocatt**
Fixes a bad formatting of the error in the cloudapi.

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] create a file in [news/unreleased](https://github.com/osbuild/osbuild-composer/tree/main/docs/news/unreleased) directory if this change should be mentioned in the release news
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

**Review by ondrejbudai — APPROVED** [review]
Good catch.

**Review by Gundersanne — APPROVED** [review]
There's a few places actually where we make this mistake.

```
server.go:146:		return echo.NewHTTPError(http.StatusBadRequest, "Unsupported distribution: %s", request.Distribution)
server.go:180:			return echo.NewHTTPError(http.StatusBadRequest, "Unsupported architecture '%s' for distribution '%s'", ir.Architecture, request.Distribution)
server.go:184:			return echo.NewHTTPError(http.StatusBadRequest, "Unsupported image type '%s' for %s/%s", ir.ImageType, ir.Architecture, request.Distribution)
server.go:216:				return echo.NewHTTPError(error_type, "Failed to depsolve base packages for %s/%s/%s: %s", ir.ImageType, ir.Architecture, request.Distribution, err)
server.go:237:			return echo.NewHTTPError(http.StatusBadRequest, "Invalid OSTree ref: %s", *ostreeOptions.Ref)
server.go:247:				return echo.NewHTTPError(http.StatusBadRequest, "Error resolving OSTree repo %s: %s", imageOptions.OSTree.URL, err)
server.go:278:			return echo.NewHTTPError(http.StatusBadRequest, "Failed to get manifest for for %s/%s/%s: %s", ir.ImageType, ir.Architecture, request.Distribution, err)
server.go:446:		return echo.NewHTTPError(http.StatusBadRequest, "Invalid format for parameter id: %s", err)
server.go:452:		return echo.NewHTTPError(http.StatusNotFound, "Job %s not found: %s", id, err)
server.go:459:			return echo.NewHTTPError(http.StatusInternalServerError, "Job %s returned more upload target results than allowed", id)
server.go:494:			return echo.NewHTTPError(http.StatusInternalServerError, "Job %s returned unknown upload target results %s", id, tr.Name)
server.go:558:		return echo.NewHTTPError(http.StatusBadRequest, "Invalid format for parameter id: %s", err)
server.go:564:		return echo.NewHTTPError(http.StatusNotFound, "Job %s not found: %s", id, err)
server.go:569:		return echo.NewHTTPError(http.StatusNotFound, "Job %s not found: %s", id, err)
```

**Comment by lavocatt** [comment]
> There's a few places actually where we make this mistake.
> 
> ```
> server.go:146:		return echo.NewHTTPError(http.StatusBadRequest, "Unsupported distribution: %s", request.Distribution)
> server.go:180:			return echo.NewHTTPError(http.StatusBadRequest, "Unsupported architecture '%s' for distribution '%s'", ir.Architecture, request.Distribution)
> server.go:184:			return echo.NewHTTPError(http.StatusBadRequest, "Unsupported image type '%s' for %s/%s", ir.ImageType, ir.Architecture, request.Distribution)
> server.go:216:				return echo.NewHTTPError(error_type, "Failed to depsolve base packages for %s/%s/%s: %s", ir.ImageType, ir.Architecture, request.Distribution, err)
> server.go:237:			return echo.NewHTTPError(http.StatusBadRequest, "Invalid OSTree ref: %s", *ostreeOptions.Ref)
> server.go:247:				return echo.NewHTTPError(http.StatusBadRequest, "Error resolving OSTree repo %s: %s", imageOptions.OSTree.URL, err)
> server.go:278:			return echo.NewHTTPError(http.StatusBadRequest, "Failed to get manifest for for %s/%s/%s: %s", ir.ImageType, ir.Architecture, request.Distribution, err)
> server.go:446:		return echo.NewHTTPError(http.StatusBadRequest, "Invalid format for parameter id: %s", err)
> server.go:452:		return echo.NewHTTPError(http.StatusNotFound, "Job %s not found: %s", id, err)
> server.go:459:			return echo.NewHTTPError(http.StatusInternalServerError, "Job %s returned more upload target results than allowed", id)
> server.go:494:			return echo.NewHTTPError(http.StatusInternalServerError, "Job %s returned unknown upload target results %s", id, tr.Name)
> server.go:558:		return echo.NewHTTPError(http.StatusBadRequest, "Invalid format for parameter id: %s", err)
> server.go:564:		return echo.NewHTTPError(http.StatusNotFound, "Job %s not found: %s", id, err)
> server.go:569:		return echo.NewHTTPError(http.StatusNotFound, "Job %s not found: %s", id, err)
> ```

Oh! Should I fix this in this PR ?

**Comment by ondrejbudai** [comment]
Depends on you, we can merge this firstly and fix others in a follow-up, or fix everything in this PR.

**Comment by lavocatt** [comment]
I'm gonna correct them. It'll be done

File: internal/cloudapi/server.go [hunk 1]
```diff
@@ -426,27 +426,27 @@ func (h *apiHandlers) Compose(ctx echo.Context) error {
 	var response ComposeResult
 	response.Id = id.String()
 
-	return ctx.JSON(http.StatusOK, response)
+	return ctx.JSON(http.StatusCreated, response)
```
Comment (Gundersanne) [inline internal/cloudapi/server.go:429]: also fixed in #1712, but it'll just rebase out

**Review by Gundersanne — APPROVED** [review]
Thank you! Good to have this cleaned up.

File: internal/cloudapi/server.go (hunk 1 above)
Comment (lavocatt) [reply internal/cloudapi/server.go:429]: Ooops, I should not have commit this.

Comment (lavocatt) [reply internal/cloudapi/server.go]: I've cleaned it. I don't want to mix apples and oranges :)

**Review by Gundersanne — APPROVED** [review]

**Review by ondrejbudai — APPROVED** [review]
Thanks!

**Pull request merged**